wp_api/src/date.rs

```diff
@@ -0,0 +1,111 @@
+use std::fmt::Display;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use chrono::DateTime;
+use chrono::NaiveDateTime;
+use chrono::TimeZone;
+use chrono::Utc;
+use serde::Deserializer;
+use serde::Serializer;
+use serde::{Deserialize, Serialize};
+
+pub const REST_API_DATE_FORMAT: &str = "%Y-%m-%dT%H:%M:%S";
+
+#[derive(Debug, Clone, PartialEq, Eq, uniffi::Record)]
+pub struct WpDateTime {
+    gmt: Arc<GMTDateTime>,
+}
+
+impl WpDateTime {
+    pub fn date_time(&self) -> &DateTime<Utc> {
+        &self.gmt.date_time
+    }
+}
+
+// This is an intermediate struct to allow the public `WpDateTime` to be exported as a
+// uniffi Record type. This struct itself is exported as a uniffi Object type.
+#[derive(Debug, Clone, PartialEq, Eq, uniffi::Object)]
+struct GMTDateTime {
```

Comment:
I don't think we need both `WpDateTime` & `GMTDateTime` at the same time since one is just a wrapper for the other. We can use one exported `uniffi::Object` with an inner type and that should be enough.